x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/use_generation.tsx

```diff
@@ -0,0 +1,224 @@
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import { useCallback, useEffect, useState } from 'react';
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { getLangSmithOptions } from '../../../../../common/lib/lang_smith';
+import type { Docs, ESProcessorItem, Pipeline, SamplesFormat } from '../../../../../../common';
+import {
+  type AnalyzeLogsRequestBody,
+  type CategorizationRequestBody,
+  type EcsMappingRequestBody,
+  type RelatedRequestBody,
+} from '../../../../../../common';
+import {
+  runCategorizationGraph,
+  runEcsGraph,
+  runRelatedGraph,
+  runAnalyzeLogsGraph,
+} from '../../../../../common/lib/api';
+import { useKibana } from '../../../../../common/hooks/use_kibana';
+import type { State } from '../../state';
+import * as i18n from './translations';
+import { useTelemetry } from '../../../telemetry';
+import type { ErrorCode } from '../../../../../../common/constants';
+import type { AIConnector, IntegrationSettings } from '../../types';
+
+export type OnComplete = (result: State['result']) => void;
+export const ProgressOrder = ['ecs', 'categorization', 'related'];
+export type ProgressItem = (typeof ProgressOrder)[number];
+
+interface UseGenerationProps {
+  integrationSettings: State['integrationSettings'];
+  connector: State['connector'];
+  onComplete: OnComplete;
+}
+
+interface RunGenerationProps {
+  integrationSettings: IntegrationSettings;
+  connector: AIConnector;
+  deps: { http: HttpSetup; abortSignal: AbortSignal };
+  setProgress: (progress: ProgressItem) => void;
+}
+
+interface GenerationResults {
+  pipeline: Pipeline;
+  docs: Docs;
+  samplesFormat?: SamplesFormat;
+}
+
+export const useGeneration = ({
+  integrationSettings,
+  connector,
+  onComplete,
+}: UseGenerationProps) => {
+  const { reportGenerationComplete } = useTelemetry();
+  const { http, notifications } = useKibana().services;
+  const [progress, setProgress] = useState<ProgressItem>();
+  const [error, setError] = useState<null | string>(null);
+  const [isRequesting, setIsRequesting] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (
+      !isRequesting ||
+      http == null ||
+      connector == null ||
+      integrationSettings == null ||
+      notifications?.toasts == null
+    ) {
+      return;
+    }
+    const generationStartedAt = Date.now();
+    const abortController = new AbortController();
+    const deps = { http, abortSignal: abortController.signal };
+
+    (async () => {
+      try {
+        const { pipeline, docs, samplesFormat }: GenerationResults = await runGeneration({
+          integrationSettings,
+          connector,
+          deps,
+          setProgress,
+        });
+        reportGenerationComplete({
+          connector,
+          integrationSettings,
+          durationMs: Date.now() - generationStartedAt,
+        });
+        const result = { pipeline, docs, samplesFormat };
+        onComplete(result);
```

Comment:
The `runGeneration` function is already very well typed, no need to explicitly type the result here. We can just do:

```suggestion
        const results = await runGeneration({ integrationSettings, connector, deps, setProgress });
        const durationMs = Date.now() - generationStartedAt;
        reportGenerationComplete({ connector, integrationSettings, durationMs });
        onComplete(result);
```

Reply:
Fixed in [f348f4e](https://github.com/elastic/kibana/pull/194150/commits/f348f4e4318f065e52cdfc3bff991b7dff409a31)